From 40959cb3e51c769060aeed1f699bf66b7770a73a Mon Sep 17 00:00:00 2001
From: Arun Ravindran <arun.ravindran@intel.com>
Date: Mon, 19 Nov 2012 16:23:06 +0200
Subject: [PORT FROM MAIN] Telephony: Add support for RIL_UNSOL_OEM_HOOK_RAW in CommandsInterface

BZ: 22242

The Android framework doesn't fully support the unsolicited notification
RIL_UNSOL_OEM_HOOK_RAW.

Change-Id: Iccc646c46de9ae1ec9bcc2f36be29b56d4c6cd54
Orig-Change-Id: I065757555d5cd02f2fe4d4bcc6c5986cf5d7a48b
Category: device enablement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: no, proprietary
Signed-off-by: Frederic Predon <frederic.predon@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../internal/telephony/CommandsInterface.java      |   11 +++++++++++
 1 files changed, 11 insertions(+), 0 deletions(-)

diff --git a/src/java/com/android/internal/telephony/CommandsInterface.java b/src/java/com/android/internal/telephony/CommandsInterface.java
index 99274f8..946947f 100644
--- a/src/java/com/android/internal/telephony/CommandsInterface.java
+++ b/src/java/com/android/internal/telephony/CommandsInterface.java
@@ -428,6 +428,17 @@ public interface CommandsInterface {
     void registerForSignalInfo(Handler h, int what, Object obj);
     void unregisterForSignalInfo(Handler h);
 
+     /**
+     * Sets the handler for OEM Hook Raw Notifications.
+     * Unlike the register* methods, there's only one notification handler
+     *
+     * @param h Handler for notification message.
+     * @param what User-defined message code.
+     * @param obj User object.
+     */
+     void setOnUnsolOemHookRaw(Handler h, int what, Object obj);
+     void unSetOnUnsolOemHookRaw(Handler h);
+
     /**
      * Registers the handler for CDMA number information record
      * Unlike the register* methods, there's only one notification handler
-- 
1.7.4.1

